torchgeo/datasets/dota.py

```diff
@@ -0,0 +1,441 @@
+    """DOTA dataset.
+
+    The `DOTA <https://captain-whu.github.io/DOTA/index.html>`_ is a large-scale object
+    detection dataset for aerial imagery containing RGB and gray-scale imagery from Google Earth, GF-2 and JL-1 satellites
+    as well as additional aerial imagery from CycloMedia.
+
+    Dataset features:
+
+    * multi-class object detection (15 classes in V1 and 18 classes in V2)
+    * horizontal and oriented bounding boxes
+
+    Dataset format:
+
+    * images are three channel PNGs with various pixel sizes
+    * annotations are text files with one line per bounding box
+
+    Classes:
+
+    * plane
+    * ship
+    * storage-tank
+    * baseball-diamond
+    * tennis-court
+    * basketball-court
+    * ground-track-field
+    * harbor
+    * bridge
+    * large-vehicle
+    * small-vehicle
+    * helicopter
+    * roundabout
+    * soccer-ball-field
+    * swimming-pool
+    * container-crane
+    * airport
+    * helipad
+
+
+    If you use this work in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2102.12219
+    * https://arxiv.org/abs/1711.10398
+
+    .. versionadded:: 0.7
+    """
+
+    url = 'https://huggingface.co/datasets/torchgeo/dota/resolve/main/{}'
+
+    file_info: ClassVar[dict[str, dict[str, dict[str, dict[str, str]]]]] = {
+        'train': {
+            'images': {
+                '1.0': {'filename': 'dotav1_images_train.tar.gz', 'md5': ''},
+                '2.0': {'filename': 'dotav2_images_train.tar.gz', 'md5': ''},
+            },
+            'annotations': {
+                '1.0': {'filename': 'dotav1_annotations_train.tar.gz', 'md5': ''},
+                '2.0': {'filename': 'dotav2_annotations_train.tar.gz', 'md5': ''},
+            },
+        },
+        'val': {
+            'images': {
+                '1.0': {'filename': 'dotav1_images_val.tar.gz', 'md5': ''},
+                '2.0': {'filename': 'dotav2_images_val.tar.gz', 'md5': ''},
+            },
+            'annotations': {
+                '1.0': {'filename': 'dotav1_annotations_val.tar.gz', 'md5': ''},
+                '2.0': {'filename': 'dotav2_annotations_val.tar.gz', 'md5': ''},
+            },
+        },
+    }
+
+    sample_df_path = 'samples.parquet'
+
+    classes = (
+        'plane',
+        'ship',
+        'storage-tank',
+        'baseball-diamond',
+        'tennis-court',
+        'basketball-court',
+        'ground-track-field',
+        'harbor',
+        'bridge',
+        'large-vehicle',
+        'small-vehicle',
+        'helicopter',
+        'roundabout',
+        'soccer-ball-field',
+        'swimming-pool',
+        'container-crane',
+        'airport',
+        'helipad',
+    )
+
+    valid_splits = ('train', 'val')
+    valid_versions = ('1.0', '2.0')
+
+    valid_orientations = ('hbb', 'obb')
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        version: str = '2.0',
+        bbox_orientation: str = 'hbb',
```

Comment:
```suggestion
        split: Literal['train', 'val'] = 'train',
        version: Literal['1.0', '2.0'] = '2.0',
        bbox_orientation: Literal['hbb', 'obb'] = 'hbb',
```
If you want to get technical, these type hints are more correct, but we don't yet use them anywhere else in TorchGeo (although we should).